### PR TITLE
Update the external load balancer test to use a different namespace in each run

### PR DESCRIPTION
**Pull request by a-robinson**
To make stuck resources less painful.

@roberthbailey @quinton-hoole

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -275,9 +275,6 @@ var _ = Describe("Services", func() {
 			},
 		}
 
-		By("cleaning up previous service " + serviceName + " from namespace " + ns)
```
Comment (roberthbailey) [inline test/e2e/service.go]: Are we going to start leaking services target pools/forwarding rules? I don't see a defer statement to clean this up and we won't automatically clear it out when deleting the cluster.

**Comment by quinton-hoole** [comment]
Note that this is essentially a duplicate of #6847, which I had to roll back, and have not yet had a chance to debug.

File: test/e2e/service.go (hunk 1 above)
Comment (a-robinson) [inline test/e2e/service.go]: See lines 284-288 - we defer deletion of the service as soon as we see that creation succeeded.

It is still possible that we could leak resources if the cluster is deleted before the service controller has a chance to delete them, but that can't really be avoided without reusing the names.

**Comment by quinton-hoole** [comment]
LGTM.  Happy to have this in.

**Comment by a-robinson** [comment]
Ah, I'm sorry Quinton. I remembered you had made that change, but didn't realize you had reverted it, assuming you just hadn't converted this specific test. Was the elb test one of the ones failing in jenkins? This definitely worked when running it on GCE manually.

The motivation for changing this one test was that the gke-ci e2e tests will continue failing until we start using a different namespace for it.

File: test/e2e/service.go (hunk 1 above)
Comment (roberthbailey) [inline test/e2e/service.go]: Sorry, it was hidden by the fold and I didn't expand it to see.

**Comment by quinton-hoole** [comment]
No need to apologise.  I should be apologising for not having fixed the bug in #6847 yet.
It was the DNS test that failed consistently on Jenkins in my case.  Passed on my local cluster though.  Slightly mysterious.

**Comment by quinton-hoole** [comment]
Can be merged as soon as travis or shippable are green.

**Pull request merged**